frontend/public/components/container.tsx

```diff
@@ -379,40 +394,60 @@ const ContainerDetails: React.FC<ContainerDetailsProps> = (props) => {
 ContainerDetails.displayName = 'ContainerDetails';
 
 export const ContainersDetailsPage: React.FC<ContainerDetailsPageProps> = (props) => {
+  return (
+    <Firehose
+      resources={[
+        {
+          name: props.match.params.podName,
+          namespace: props.match.params.ns,
+          kind: 'Pod',
+          isList: false,
+          prop: 'obj',
+        } as FirehoseResource,
+      ]}
+    >
+      <ContainersDetailsPage_ {...props} />
+    </Firehose>
+  );
+};
+
+export const ContainersDetailsPage_: React.FC<ContainerDetailsPageProps> = (props) => {
   const { t } = useTranslation();
+  const pod = props.obj.data;
+  const container = getContainer(pod, props.match.params.name);
+
+  if (!container) {
+    return null;
```

Comment:
If the pod is loaded and there is no container, I would expect us to show a 404 page.

Reply:
Thanks for the extensive feedback! Is there already a specific component, which should be used for displaying a 404 page?